### PR TITLE
Allows supermatter crystal to produce gases on non-space turfs that don't have air.

### PR DESCRIPTION
**Pull request by Pickle-Coding**
## About The Pull Request
If there is no air on a turf, the supermatter crystal will not produce gases even if powered. This changes that so that it can produce gases while powered if there are none on the turf. It will still take damage in case it doesn't produce gases, or it gets immediately removed.

## Why It's Good For The Game
Although it's unlikely to happen, it will look very weird if the supermatter doesn't produce gas while on a non-space turf while powered, as that situation but with like 0.0001 moles on the turf would cause the SM to immediately go on fire.

## Changelog

:cl:
fix: Allows the supermatter crystal to produce gases while powered, even in absolutely empty turfs, excluding space turfs.
/:cl:

### CODE REVIEW

**Comment by Pickle-Coding** [comment]
I thought I rebased it, but I guess not. Sad.

File: code/modules/power/supermatter/supermatter.dm [hunk 1]
```diff
@@ -539,6 +539,20 @@ GLOBAL_DATUM(main_supermatter_engine, /obj/machinery/power/supermatter_crystal)
 	if(!removed || !removed.total_moles() || isspaceturf(local_turf)) //we're in space or there is no gas to process
 		if(takes_damage)
 			damage += max((power / 1000) * DAMAGE_INCREASE_MULTIPLIER, 0.1) // always does at least some damage
+		if(!isspaceturf(local_turf) && produces_gas && power) //There is no gas to process, but we are not in a space turf. Lets make them.
```
Comment (LemonInTheDark) [inline code/modules/power/supermatter/supermatter.dm:542]: could you go based off the immutability of the gas mixture, rather then the turf being a space tile?

Comment (Pickle-Coding) [reply code/modules/power/supermatter/supermatter.dm]: Yes.

**Comment by LemonInTheDark** [comment]
Shouldn't mention immutable in the changelog, players don't know what it means. Maybe just space?

Outside of that this is good to go

File: code/modules/power/supermatter/supermatter.dm [hunk 2]
```diff
@@ -711,7 +725,7 @@ GLOBAL_DATUM(main_supermatter_engine, /obj/machinery/power/supermatter_crystal)
 		//Power * 0.55 * (some value between 1.5 and 23) / 5
 		removed.temperature += ((device_energy * dynamic_heat_modifier) / THERMAL_RELEASE_MODIFIER)
 		//We can only emit so much heat, that being 57500
-		removed.temperature = max(0, min(removed.temperature, 2500 * dynamic_heat_modifier))
+		removed.temperature = max(TCMB, min(removed.temperature, 2500 * dynamic_heat_modifier))
```
[the remark this answers is not in the document]
Comment (LemonInTheDark) [reply code/modules/power/supermatter/supermatter.dm:728]: Mood (If I die in a car accident, look at ice peppers)

**Comment by Pickle-Coding** [comment]
Done.

**Pull request merged**